src/domain.rs

```diff
@@ -181,6 +188,54 @@ macro_rules! unique_domain {
     }};
 }
 
+/// Generate a static [`Domain`] with an entirely unique domain family.
+///
+/// Usage: `unique_domain_family!(static DOMAIN: Domain<Family>);`
+///
+/// This macro is useful when you want to store a domain in a static variable, and makes it
+/// possible to name the Domain. The generated family implements [`Singleton`], which enables
+/// the use of [`AtomicPtr::safe_load`](crate::AtomicPtr::safe_load). This means it's impossible to
+/// construct an instance of Family outside of this macro, despite the ability to name the Family
+/// Type.
+///
+/// This is useful since it allows you to store the [`HazardPointer`]s aquired from this domain, since
+/// the Family type can now be named.
+///
+/// ```rust
+/// # struct DataStructure;
+/// # use haphazard::{HazardPointer, AtomicPtr, static_unique_domain};
+/// static_unique_domain!(static LOCAL: Domain<Local>);
+///
+/// struct ContainsHazardPointers<'domain> {
+///     haz_ptr: HazardPointer<'domain, Local>,
+///     val: AtomicPtr<DataStructure, Local>,
+/// }
+///
+/// impl ContainsHazardPointers<'_> {
+///     fn read(&mut self) -> Option<&DataStructure> {
+///         self.val.safe_load(&mut self.haz_ptr)
+///     }
+/// }
+/// # fn main() {}
+/// ```
+#[macro_export]
+macro_rules! static_unique_domain {
+    ($v:vis static $domain:ident: Domain<$family:ident>) => {
+        struct UniqueFamily;
+        // Safety: nowhere else can construct an instance of UniqueFamily to pass to Domain::new.
```

Comment:
Unfortunately, I don't think this is true. As you can see from [this playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=4cbb503bffe2921fb9f9adf9331839c8). The reason for this is that macros are only hygienic with regards to _variables_, not with regards to _types_. So `UniqueFamily` "leaks" out through the macro. You can also see that in that this macro cannot be invoked more than once in a single file because they'll both declare the `UniqueFamily` type.

Reply:
This is actually also an issue with the current `unique_domain!` macro - I don't know if it slipped through a review at some point, but the type defined by the `unique_domain!` macro is also leaked: [see this playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=4cbb503bffe2921fb9f9adf9331839c8). 

Obviously, the original solution for `unique_domain!` (making the type a lambda expression) should work, but this might make it impossible to implement a `static_unique_domain!`.